### PR TITLE
Fix a bug in the `/event_reports` Admin API which meant that the total count could be larger than the number of results you can actually query for.

### PR DESCRIPTION
**Pull request by reivilibre**
The two queries:

https://github.com/matrix-org/synapse/blob/583f22780f44157c50bc2dc5c242e88cc18c7886/synapse/storage/databases/main/room.py#L2004-L2037

need to match in both `WHERE` conditions and `JOIN` conditions (insofaras the joins can change the number of rows returned; the `LEFT JOIN` is safe because it won't remove any rows and neither will it lead to more rows being produced, since only one event per event ID can exist.)

### CODE REVIEW

**Review by babolivier — COMMENTED** [review]
Is there any test for this endpoint? It looks like something we want to write a regression test for.

**Comment by DMRobertson** [comment]
Looks like #13521 has made the linting unhappy. I think the problem is that we're restoring a mypy 0.950 cache and mypy 0.971 is unhappy with it (or a similar problem with mypy-zope). Could try re-running this? Otherwise I guess it's time to fork/file a bug under https://github.com/AustinScola/mypy-cache-github-action

**Comment by reivilibre** [comment]
> Is there any test for this endpoint? It looks like something we want to write a regression test for.

happy to add a test but it also feels like the kind of test that's a bit trivial / not really testing anything once it's fixed, but ah well

**Comment by babolivier** [comment]
> not really testing anything once it's fixed

The point of a regression test is that it should only fail if we accidentally break it in the future (e.g. if we change the query to get the results but forget to change the one to get the count), so I would say it's still a helpful thing to have.

File: tests/rest/admin/test_event_reports.py [hunk 1]
```diff
@@ -410,6 +410,35 @@ def _check_fields(self, content: List[JsonDict]) -> None:
             self.assertIn("score", c)
             self.assertIn("reason", c)
 
+    def test_count_correct_despite_table_deletions(self) -> None:
+        """
+        Tests that the issue fixed by https://github.com/matrix-org/synapse/pull/13525
+        does not recur:
+        the count should match the number of rows, even if rows in joined tables
+        are missing.
```
Comment (babolivier) [inline tests/rest/admin/test_event_reports.py:418]:
```suggestion
        Tests that the count matches the number of rows, even if rows in joined tables
        are missing.
```

Not sure it's worth mentioning the PR in the docstring. I'm not feeling incredibly strongly about it though so happy to keep it if you disagree.

**Review by babolivier — APPROVED** [review]

**Pull request merged**